transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -408,6 +409,11 @@ public final ChannelPromise voidPromise() {
         return pipeline.voidPromise();
     }
 
+    final void autoFlushModified(boolean autoFlush) {
```

Comment:
this would require adding a method to an `ChannlConfig` but would get rid of all the coupling introduced by this PR ...  what if autoFlush changing from false to true did `channel.write(EMPTY_BUFFER)` and then the channel could just inspect `config.autoFlush()`?